pkg/upgrade/upgrader/helper.go

```diff
@@ -307,3 +311,10 @@ func buildUpgradeTask(kind, name, openebsNamespace string) *apis.UpgradeTask {
 	}
 	return utaskObj
 }
+
+func verifyDeploymentReplicaStatus(d *appsv1.Deployment) error {
+	if d.Status.ReadyReplicas != *d.Spec.Replicas {
+		return errors.New(d.Name + " deployment pod is not in running state.")
```

Comment:
better to give more info like ReadyReplicasCount and Spec.Replicas as well

Reply:
think of doing nil check and then moving to k8s deployment package

Reply:
Moved the function to deployment package.